### ISSUE TITLE
Pasting in code block erases text in IceCat & IceWeasel (Firefox)

### ISSUE THREAD

**Issue by mind-overflow** [ProseMirror/prosemirror#1240]
Pasting in code-blocks erases text preceding what is pasted.  
  
**To Reproduce**
1. open [ProseMirror Demo](https://prosemirror.net/examples/basic/)
2. create a code block
3. write something into the block
4. select something and copy-paste it in the code block itself
You will notice that by pasting the text, it also erases what was preceding it, only leaving the next part.

**Expected behavior**
It should just paste the text where the cursor is, like it does in other browsers or versions.

**Screenshots**
I uploaded a short video.
*Notes:*
- *this video was recorded on Outline, which is not ProseMirror itself, but instead integrates it. However, this is perfectly reproducible in the ProseMirror Demo too, which I tried before opening this issue.*
- *I had a few extensions installed at time of recording, but even after disabling all of them, the issue persists.*
- *I also tried changing security settings (third-party cookies, HTTPS everywhere, DoNotTrack, ...) to no avail.*

**Desktop:**
- OS: Windows 11 on Surface Pro 8
- Browser: IceCat 91.5.0 ESR and IceWeasel 98.0

This issue has been copied from outline/outline#3065, where I first reported it. I highly suggest also reading the few comments there, as there is additional info that could potentially help investigating.  
  
Thank you!

**Comment by marijnh** [comment]
This does not seem to happen with stock Firefox (97, Linux).

**Comment by mind-overflow** [comment]
No, indeed it does not happen with stock Firefox. I wrote Firefox in the title to clarify that those browsers are based on Firefox and not Chromium, in case anyone didn't know them.

**Comment by marijnh** [comment]
This issue has been moved to [issue #1240](https://code.haverbeke.berlin/prosemirror/prosemirror/issues/1240) on the [repository's new home](https://code.haverbeke.berlin/prosemirror/prosemirror/).
